### PR TITLE
Remove sidecars with invalid proofs

### PR DESCRIPTION
**Pull request by potuz**
Remove sidecars when they have invalid proofs. Also rename function in anticipation of an existing DeleteBlobSidecar

### CODE REVIEW

File: beacon-chain/blockchain/process_block.go [hunk 1]
```diff
@@ -564,6 +564,8 @@ func (s *Service) isDataAvailable(ctx context.Context, root [32]byte, signed int
 		if len(sidecars) >= expected {
 			s.blobNotifiers.delete(root)
 			if err := kzg.IsDataAvailable(kzgCommitments, sidecars); err != nil {
+				log.WithField("root", fmt.Sprintf("%#x", root)).Warn("removing blob sidecars with invalid proofs")
```
Comment (potuz) [inline beacon-chain/blockchain/process_block.go:567]: There are two possible errors, the length of commitments is not right or there is an invalid proof. It's safe to remove blobs anyway here.

**Review by kasey — DISMISSED** [review]

**Review by terencechain — CHANGES_REQUESTED** [review]
Instead if merging this. Why not have Devnet nodes run from this branch for now? And we can merge in the real fixes in a few days?

**Review by terencechain — APPROVED** [review]
Discussed offline. Ok to merge now while we work on the cache. Thanks Potuz!

**Review by terencechain — COMMENTED** [review]
Reviewing this more, you'll want to remove the sidecars in the for loop to save 4 seconds

File: beacon-chain/blockchain/process_block.go [hunk 2]
```diff
@@ -564,6 +564,10 @@ func (s *Service) isDataAvailable(ctx context.Context, root [32]byte, signed int
...
+				if err2 := s.cfg.BeaconDB.DeleteBlobSidecars(ctx, root); err2 != nil {
```
[the remark this answers is not in the document]
Comment (nisdas) [reply beacon-chain/blockchain/process_block.go:568]: maybe rename this to `deletionErr` ? we don't want to overwrite the original error as it still has to be returned to the caller

[the remark this answers is not in the document]
Comment (terencechain) [reply beacon-chain/blockchain/process_block.go:568]: Sorry I missed that

**Review by terencechain — APPROVED** [review]

**Pull request merged**